tests/RPM/TagsSpec.hs

```diff
@@ -0,0 +1,26 @@
+module RPM.TagsSpec (main, spec) where
+
+import Test.Hspec
+import RPM.Tags(mkTag, Tag(..))
+import qualified Data.ByteString as BS
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+spec = describe "RPM.Tags" $
+  describe "mkTag" $ do
+    it "returns Nothing for unknown `tag' value" $ do
+      let store = BS.pack [0,0,0,5]
+      let tag = mkTag store 9999 0 0 0
+      tag `shouldBe` Nothing
+
+    it "returns FileModes for 1030 `tag' value" $ do
+      let store = BS.pack [0,0,0,5]
+      let tag = mkTag store 1030 3 0 2
+      tag `shouldBe` Just (FileModes [0,5])
+
+    it "returns Nothing for 1030 `tag' value and `ty' != 3" $ do
+      let store = BS.pack [0,0,0,5]
+      let tag = mkTag store 1030 0 0 2
+      tag `shouldBe` Nothing
```

Comment:
I can do a loop over different tag values which are handled in the same way and then define the hspec examples for `x` :).

I will checkout the RPM code to get an idea about meaningful values to use for testing. Otherwise I will be testing with zeroes most likely.